src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,176 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.InvalidIndexException;
+
+import duke.ui.Span;
+
+import duke.task.Deadline;
+import duke.task.ToDo;
+import duke.task.TaskList;
+import duke.task.Event;
+import duke.task.Task;
+
+/**
+ * Parser to parse the command and process the necessary actions associated with the command.
+ *
+ * @author Andre Lin HuiKai
+ * @version CS2103T AY22/23 Semester 2
+ */
+public class Parser {
+    private TaskList tasks;
+
+    /**
+     * Constructor for parser.
+     * @param tasks TaskList of tasks for the Parser to retrieve the tasks and act on it accordingly.
+     */
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     *
+     * @param command
+     * @param echo The user input to be parsed to be used in the execution of its associated command.
+     * @return a boolean value that determines whether the program will continue or halt.
+     * @throws DukeException DukeException thrown if there is missing required information in the user's input.
+     */
+    public boolean execute(Command command, String echo) throws DukeException {
```

Comment:
Should the various types of commands be abstracted out into different classes, so as to make use of dynamic binding?

Reply:
Yes for extensibility. But i decided against it because 
1. takes time 
2. The initial parsing using regex differs for each command and i can't see much commonalities among the commands at least up till the current context.
Thanks!